libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationObserver.kt

```diff
@@ -0,0 +1,75 @@
+package com.mapbox.navigation.core.lifecycle
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * This allows you to attach and detach [MapboxNavigation] observers
+ * from any component with a lifecycle.
+ *
+ * Example of observing locations with a view model
+ * ```
+ * class MyViewModel : ViewModel() {
+ *   private val locationObserver = MyLocationObserver()
+ *
+ *   val location: LiveData<Location> = locationObserver.location.asLiveData()
+ *
+ *   init {
+ *     MapboxNavigationApp.register(locationObserver)
+ *   }
+ *
+ *   override fun onCleared() {
+ *     MapboxNavigationApp.unregister(locationObserver)
+ *   }
+ * }
+ *
+ * class MyLocationObserver : MapboxNavigationObserver {
+ *   private val mutableLocation = MutableStateFlow<LocationMatcherResult?>(null)
+ *   val locationFlow: Flow<LocationMatcherResult?> = mutableLocation
+ *
+ *   private val locationObserver = object : LocationObserver {
+ *     override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
+ *       mutableLocation.value = locationMatcherResult
+ *     }
+ *
+ *     override fun onNewRawLocation(rawLocation: Location) {
+ *       // no op
+ *     }
+ *   }
+ *
+ *   override fun onAttached(mapboxNavigation: MapboxNavigation) {
+ *     mapboxNavigation.registerLocationObserver(locationObserver)
+ *   }
+ *
+ *   override fun onDetached(mapboxNavigation: MapboxNavigation) {
+ *     mapboxNavigation.unregisterLocationObserver(locationObserver)
+ *   }
+ * }
+ * ```
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+interface MapboxNavigationObserver {
+    /**
+     * After you have registered an observer through [MapboxNavigationApp.registerObserver],
+     * the onAttached will be called when at least one [LifecycleOwner] that has been attached to
+     * [MapboxNavigationApp.attach] is at least in the [Lifecycle.State.STARTED] state.
+     */
+    fun onAttached(mapboxNavigation: MapboxNavigation)
+
+    /**
+     * There are a few scenarios which can cause onDetached to be called. You can assume that this
+     * will only be called if [onAttached] has been called. The [mapboxNavigation] instance will
+     * be the same instance as the one passed to the [onAttached].
+     *
+     * This is called after all [LifecycleOwner]s that have been attached to
```

Comment:
Do _all_ owners need to be started or rather _any_ owner?

Reply:
Yeah it's all lifecycle owners need to be stopped. Allowing for car and app. 

Future iteration here of `MapboxNavigationApp`, or an extension of the singleton, will be to hook a MapboxNavigationObserver to a specific LifecycleOwner.

Reply:
Updating the description. But may need to add something to address this expectation